internal/commands/app/create_test.go

```diff
@@ -335,75 +330,59 @@ Check out your app: cd ./test-app && realm-cli app describe
-		assert.Nil(t, err)
-		defer zipPkg.Close()
-
-		rc.ExportFn = func(groupID, appID string, req realm.ExportRequest) (string, *zip.Reader, error) {
-			return "", &zipPkg.Reader, err
-		}
-		rc.CreateAppFn = func(groupID, name string, meta realm.AppMeta) (realm.App, error) {
-			createdApp = realm.App{
-				GroupID:     groupID,
-				ID:          "456",
-				ClientAppID: name + "-abcde",
-				Name:        name,
-				AppMeta:     meta,
-			}
-			return createdApp, nil
-		}
-		rc.ImportFn = func(groupID, appID string, appData interface{}) error {
-			return nil
-		}
-		rc.AllTemplatesFn = func() ([]realm.Template, error) {
-			return []realm.Template{
-				{
-					ID:   "palm-pilot.bitcoin-miner",
-					Name: "Mine bitcoin on your Palm Pilot from the comfort of your home, electricity not included",
-				},
-				{
-					ID:   "blackberry.important-business-app",
-					Name: "Oh wow, a Blackberry... you must a very powerful, extravagant man.",
-				},
-			}, nil
+		testApp := realm.App{
+			ID:          "456",
+			GroupID:     "123",
+			ClientAppID: "bitcoin-miner-abcde",
+			Name:        "bitcoin-miner",
 		}
 
-		clientZipPkg, err := zip.OpenReader("testdata/react-native.zip")
+		backendZipPkg, err := zip.OpenReader("testdata/project.zip")
 		assert.Nil(t, err)
-		rc.ClientTemplateFn = func(groupID, appID, templateID string) (*zip.Reader, bool, error) {
-			return &clientZipPkg.Reader, true, nil
-		}
-		ac := mock.AtlasClient{}
-		ac.GroupsFn = func() ([]atlas.Group, error) {
-			return []atlas.Group{{ID: "123"}}, nil
-		}
-		ac.ClustersFn = func(groupID string) ([]atlas.Cluster, error) {
-			return []atlas.Cluster{{Name: "test-cluster"}}, nil
+
+		client := mock.RealmClient{
+			FindAppsFn: func(filter realm.AppFilter) ([]realm.App, error) {
+				return []realm.App{}, nil
+			},
+			ExportFn: func(groupID, appID string, req realm.ExportRequest) (string, *zip.Reader, error) {
+				return "", &backendZipPkg.Reader, nil
+			},
+			CreateAppFn: func(groupID, name string, meta realm.AppMeta) (realm.App, error) {
+				return realm.App{
+					GroupID:     groupID,
+					ID:          "456",
+					ClientAppID: name + "-abcde",
+					Name:        name,
+					AppMeta:     meta,
+				}, nil
+			},
+			ImportFn: func(groupID, appID string, appData interface{}) error {
+				return nil
+			},
 		}
 
 		cmd := &CommandCreate{createInputs{
 			newAppInputs: newAppInputs{
-				Name:            "template-app",
-				Location:        realm.LocationVirginia,
+				Name:            "bitcoin-miner",
+				Project:         testApp.GroupID,
+				Location:        realm.LocationIreland,
 				DeploymentModel: realm.DeploymentModelGlobal,
-				ConfigVersion:   realm.DefaultAppConfigVersion,
 			},
 			Clusters: []string{"test-cluster"},
 		}}
 
-		assert.Nil(t, cmd.Handler(profile, ui, cli.Clients{Realm: rc, Atlas: ac}))
-
-		console.Tty().Close() // flush the writers
-		<-doneCh              // wait for procedure to complete
```

Comment:
is this test no longer interactive (i.e. is `procedure` needed at all)?

if its no longer interactive, we should switch the `ui` to just be `mock.NewUI` (rather than the `V10XConsole` variant), as well as this bit:

```go
		doneCh := make(chan (struct{}))
		go func() {
			defer close(doneCh)
			procedure(console)
		}()
```

if it _is_ interactive still, then we definitely should still keep the `<-doneCh` and psuedo-terminal `Close` calls

Reply:
i suppose it's because the structure of the test cases are similar, but the diff is trying to compare a deleted test case with a different one altogether